### PR TITLE
Document containers.conf settings for remote connections

### PR DESCRIPTION
**Pull request by rhatdan**
Currently we don't document which end of the podman-remote client server
operations uses the containers.conf.  This PR begins documenting this
and then testing to make sure the defaults follow the rules.

Fixes: https://github.com/containers/podman/issues/7657

Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

File: docs/source/markdown/podman-create.1.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ any point.
 
 The initial status of the container created with **podman create** is 'created'.
 
+Default settings for flags are defined in `containers.conf`. Most settings for
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:21]:
```suggestion
Default settings for options are defined in `containers.conf`. Most settings for
```

File: docs/source/markdown/podman-create.1.md [hunk 2]
```diff
@@ -893,10 +898,12 @@ standard input.
 **--tz**=*timezone*
 
 Set timezone in container. This flag takes area-based timezones, GMT time, as well as `local`, which sets the timezone in the container to match the host machine. See `/usr/share/zoneinfo/` for valid timezones.
+Remote connections, use local containers.conf for defaults
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:901]:
```suggestion
Remote connections, use local containers.conf for defaults.
```

File: docs/source/markdown/podman-create.1.md [hunk 3]
```diff
@@ -893,10 +898,12 @@ standard input.
...
 
 **--umask**=*umask*
 
 Set the umask inside the container. Defaults to `0022`.
+Remote connections, use local containers.conf for defaults
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-create.1.md:906]:
```suggestion
Remote connections, use local containers.conf for defaults.
```

File: docs/source/markdown/podman-create.1.md (hunk 1 above)
Comment (TomSweeneyRedHat) [reply docs/source/markdown/podman-create.1.md:21]: Down below you have "Default setting are defined..." If you wanted to drop "for flags" here entirely, that would work too.

File: docs/source/markdown/podman-run.1.md [hunk 4]
```diff
@@ -952,10 +957,12 @@ standard input.
 **--tz**=*timezone*
 
 Set timezone in container. This flag takes area-based timezones, GMT time, as well as `local`, which sets the timezone in the container to match the host machine. See `/usr/share/zoneinfo/` for valid timezones.
+Remote connections, use local containers.conf for defaults
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:960]:
```suggestion
Remote connections, use local containers.conf for defaults.
```

File: docs/source/markdown/podman-run.1.md [hunk 5]
```diff
@@ -952,10 +957,12 @@ standard input.
...
 
 **--umask**=*umask*
 
 Set the umask inside the container. Defaults to `0022`.
+Remote connections, use local containers.conf for defaults
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman-run.1.md:965]:
```suggestion
Remote connections, use local containers.conf for defaults.
```

File: docs/source/markdown/podman.1.md [hunk 6]
```diff
@@ -17,6 +17,10 @@ Podman uses Buildah(1) internally to create container images. Both tools share i
 (not container) storage, hence each can use or manipulate images (but not containers)
 created by the other.
 
+Default settings for flags are defined in `containers.conf`. Most settings for
```
Comment (TomSweeneyRedHat) [inline docs/source/markdown/podman.1.md:20]:
```suggestion
Default settings for options are defined in `containers.conf`. Most settings for
```
or drop for flags entirely.

**Comment by TomSweeneyRedHat** [comment]
@rhatdan test aren't hip

**Review by TomSweeneyRedHat — COMMENTED** [review]
LGTM

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
@vrothberg We talked in a different issue about adding some tests to make sure that the server side is using containers.conf,  This PR begins that and fixes some issues, where we were relying on the client side and ignoring the server side.

**Comment by vrothberg** [comment]
On my review list for tomorrow. Brain isn't sharp enough to review so many line changes.

File: pkg/specgen/volumes.go [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+	// Source is the source of the image volume.  The image can be referred
+	// to by name and by ID.
+	Source string
+	// Destination is the absolute path of the mount in the container.
+	Destination string
+	// ReadWrite sets the volume writable.
+	ReadWrite bool
+}
+
+func GenVolumeMounts(volumeFlag []string) (map[string]spec.Mount, map[string]*NamedVolume, map[string]*OverlayVolume, error) {
```
Comment (vrothberg) [inline pkg/specgen/volumes.go:49]: Doc comments are missing.

File: cmd/podman/common/create.go [hunk 8]
```diff
@@ -791,3 +791,66 @@ func DefineCreateFlags(cmd *cobra.Command, cf *ContainerCLIOpts) {
 	_ = cmd.RegisterFlagCompletionFunc(cgroupConfFlagName, completion.AutocompleteNone)
 
 }
+
+func ulimits() []string {
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply cmd/podman/common/create.go:795]: Rather then create a fourth create*.go file, how about I move them to create_opts.go.

File: docs/source/markdown/podman-create.1.md [hunk 9]
```diff
@@ -817,6 +821,7 @@ Signal to stop a container. Default is SIGTERM.
 #### **--stop-timeout**=*seconds*
 
 Timeout (in seconds) to stop a container. Default is 10.
+Remote connections, use local containers.conf for defaults
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/source/markdown/podman-create.1.md:824]: fixed

File: docs/source/markdown/podman-create.1.md [hunk 10]
```diff
@@ -18,6 +18,10 @@ any point.
 
 The initial status of the container created with **podman create** is 'created'.
 
+Default settings for flags are defined in `containers.conf`. Most settings for
+Remote connections use the server's containers.conf, except when documented in
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply docs/source/markdown/podman-create.1.md:22]: fixed

File: docs/source/markdown/podman-create.1.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (vrothberg) [reply docs/source/markdown/podman-create.1.md]: Some cases are left

File: cmd/podman/common/create.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (vrothberg) [reply cmd/podman/common/create.go]: SGTM

File: pkg/specgen/volumes.go [hunk 11]
```diff
@@ -0,0 +1,148 @@
+package specgen
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/containers/buildah/pkg/parse"
```
[the remark this answers is not in the document]
Comment (rhatdan) [reply pkg/specgen/volumes.go:7]: Will do this in a separate PR.

[the remark this answers is not in the document]
Comment (vrothberg) [reply pkg/specgen/volumes.go:7]: Awsome, thanks!

File: pkg/specgen/volumes.go (hunk 7 above)
Comment (rhatdan) [reply pkg/specgen/volumes.go:50]: Fixed.

File: cmd/podman/common/create.go (hunk 8 above)
[the remark this answers is not in the document]
Comment (rhatdan) [reply cmd/podman/common/create.go]: fixed

**Comment by rhatdan** [comment]
@giuseppe @mheon @edsantiago Any idea what is going on here. Ubuntu (cgroupv1 pod_stats_test,go is blowing up, even though these tests were not touched.

[+1082s] Error: unable to obtain cgroup stats: open /sys/fs/cgroup/pids/pids.current: no such file or directory

Have you ever seen anything like this before?

**Comment by edsantiago** [comment]
I just saw and reported [the same thing in 8170](https://storage.googleapis.com/cirrus-ci-6707778565701632-fcae48/artifacts/containers/podman/5933768118632448/html/int-remote-ubuntu-19-root-host.log.html). Pressing Re-run made it go away, which does not fill me with warm fuzzies.

**Review by vrothberg — COMMENTED** [review]
LGTM

**Review by saschagrunert — APPROVED** [review]
/lgtm

**Pull request merged**